### DISCUSSION TITLE
Custom Button which can check a checkbox and trigger the submit form action.

### DISCUSSION THREAD

**Discussion by FactorD** [team-forms/community#20] [Q&A] [answered]
Hi everyone, 

We need a custom Submit button which on-click checks a checkbox and submits the Form. 

We want to use that checkbox as a conditional to hide the first page during future edits of the Form.

Thank you!

**Comment by AnthonyPhan** [comment, marked as answer]
Hi @FactorD 

The following steps should achieve what your want :)

1. Drag and Drop a button component into your form to act as the custom submit button. 
2. For the "action" property of your button select "custom"
3. Enter the following JavaScript code that will update the value of your checkbox then submit the form. Note that you will need to update "checkboxProperty" to be the property name of your checkbox field (located in the API tab of your component) 

```
instance.root.getComponent('checkboxProperty').setValue(true);
instance.emit('submitButton')
```

**Reply by FactorD** [reply]
Thanks, Anthony! Works like a charm! :)
